contrib/analytics/ivre_api/ivre_api.py

```diff
@@ -0,0 +1,421 @@
+
+From `CertificateSubject` observables to:
+
+  - (other) `Certificate` observables, based on X509 certificates seen
+    in TLS traffic.
+
+Please refer to IVRE's documentation on how to collect passive data.
+
+    """
+
+    default_values = {
+        "name": "IVRE - Passive",
+        "description": __doc__,
+    }
+
+    ACTS_ON = ["Ip", "Hostname", "CertificateSubject"]
+
+    @classmethod
+    def analyze(cls, observable, results):
+        LOG.debug(
+            'Begin analyze %s for %r (type %s)', cls.__name__, observable,
+            observable.__class__.__name__)
+        if isinstance(observable, Ip):
+            return cls.analyze_ip(observable, results)
+        if isinstance(observable, Hostname):
+            return cls.analyze_hostname(observable, results)
+        if isinstance(observable, CertificateSubject):
+            return cls.analyze_certsubj(observable, results)
+        LOG.warning(
+            '%s: cannot analyze, unknown observable %r (type %s)', cls.__name__,
+            observable, observable.__class__.__name__)
+        return []
+
+    @classmethod
+    def analyze_ip(cls, ip, results):
+        """Specific analyzer for Ip observables."""
+
+        links = set()
+        result = {}
+        for rec in db.passive.get(db.passive.searchhost(ip.value)):
+            LOG.debug('%s.analyze_ip: record %r', cls.__name__, rec)
+            if rec["recontype"] == "DNS_ANSWER":
+                value = rec['value']
+                hostname = Hostname.get_or_create(value=value)
+                rec_type = "dns-%s" % rec["source"].split('-', 1)[0]
+                result.setdefault(rec_type, set()).add(value)
+                links.update(
+                    ip.link_to(
+                        hostname,
+                        rec_type,
+                        'IVRE - DNS-%s' % rec["source"],
+                        first_seen=rec['firstseen'],
+                        last_seen=rec['lastseen'],
+                    ))
+
+            elif rec["recontype"] == "HTTP_CLIENT_HEADER_SERVER":
+                if rec["source"] == "HOST":
+                    value = rec['value']
+                    result.setdefault('http-host', set()).add(value)
+                    _try_link(
+                        links,
+                        ip,
+                        Hostname,
+                        value,
+                        "http-host",
+                        'IVRE - HTTP Host: header',
+                        first_seen=rec['firstseen'],
+                        last_seen=rec['lastseen'])
+                else:
+                    continue
+            elif rec["recontype"] == "HTTP_SERVER_HEADER":
+                if rec["source"] == 'SERVER':
+                    value = rec['value']
+                    result.setdefault('http-server', set()).add(value)
+                    links.update(
+                        ip.link_to(
+                            Text.get_or_create(value=value),
```

Comment:
What's `value` here exactly? Is it the HTTP server's headers or one header in particular?

Reply:
In that case, that will be the `Server:` header from an HTTP server. In IVRE, interesting HTTP headers that are sent by the server are stored as a `"recontype": "HTTP_SERVER_HEADER"` record. The `"source"` field of such records is the (uppercase) header name.

Reply:
I see. Wouldn't it be interesting to switch to `Hostname` observables if it turns out the value is an actual hostname? (mostly in the case of the HTTP client headers I suppose)

Reply:
I'm not sure to understand. Hostnames are seen in `"HTTP_CLIENT_HEADER_SERVER"` record types (the name is confusing, that is what I use for HTTP headers sent by the client that provide information about the server) `"HOST"`. They are indeed used in this PR, see https://github.com/yeti-platform/yeti/pull/333/files#diff-dbdf09715f287b1e95399459faf8813aR288.

Reply:
Yeah, I'm just wondering if in "most" cases, http-post will be an actual host that is worth registering somewhere as a Hostname observable